appserver/payara-appserver-modules/microprofile/jwt-auth/src/main/java/fish/payara/microprofile/jwtauth/eesecurity/SignedJWTIdentityStore.java

```diff
@@ -121,15 +131,16 @@ public CredentialValidationResult validate(SignedJWTCredential signedJWTCredenti
             JsonWebTokenImpl jsonWebToken
                     = jwtTokenParser.verify(acceptedIssuer, publicKey.get());
 
-            List<String> groups = new ArrayList<>(
-                    jsonWebToken.getClaim("groups"));
+            Set<String> groups = new HashSet<>();
+            Collection<String> groupClaims = jsonWebToken.getClaim("groups");
+            if (groupClaims != null) {
+                groups.addAll(groupClaims);
+            }
 
-            return new CredentialValidationResult(
-                    jsonWebToken,
-                    new HashSet<>(groups));
+            return new CredentialValidationResult(jsonWebToken, groups);
 
         } catch (Exception e) {
-            LOGGER.log(FINEST, "Exception trying to parse JWT token.", e);
+            LOGGER.log(INFO, "Exception trying to parse JWT token.", e);
```

Comment:
I wanted to ask for feedback on this actually. I did this because I noted that whenever a request was unsuccessful it wasn't clear to me why without debugging. This said, do you think this will be a little too spammy?

Reply:
I know very little about the context here - would it log with basically every request of a session that had this problem or is it more once when trying to login?

Reply:
I think that it should only trigger when a JWT Token is provided, so I don't think it's too invasive. I just thought it would be beneficial to see why a token is being rejected